src/ucp/api/ucp.h

```diff
@@ -3386,32 +3411,65 @@ ucp_atomic_fetch_nb(ucp_ep_h ep, ucp_atomic_fetch_op_t opcode,
 
 /**
  * @ingroup UCP_COMM
- * @brief Post an atomic fetch operation.
+ * @brief Post an atomic memory operation.
  *
- * This routine will post an atomic fetch operation to remote memory.
+ * This routine will post an atomic operation to remote memory.
  * The remote value is described by the combination of the remote
  * memory address @a remote_addr and the @ref ucp_rkey_h "remote memory handle"
- * @a rkey.
- * The routine is non-blocking and therefore returns immediately. However the
- * actual atomic operation may be delayed. The atomic operation is not considered complete
- * until the values in remote and local memory are completed.
+ * @a rkey. The routine is non-blocking and therefore returns immediately.
+ * However the actual atomic operation may be delayed.
+ * If the reply buffer is not provided in @a param->reply_buffer, the request
+ * handle is not returned and the user must call @ref ucp_ep_flush_nb or
```

Comment:
this may not be true, if we haze zero copy on sender

Reply:
yes, but it reflects the current state. We do not provide request to the user (even if we create one)

Reply:
i think better document in a more generic way. so user will not have to change the code. if we document that we **may** return a request, we have the liberty to return/not return it in future implementation.

Reply:
(unresolved)